projects/commudle-admin/src/app/feature-modules/events/components/user-event-registrations/entry-pass-scan/entry-pass-scan.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, ElementRef, OnInit, ViewChild } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { BarcodeFormat } from '@zxing/library';
+import { ZXingScannerComponent } from '@zxing/ngx-scanner';
+import { EventPassService } from 'projects/commudle-admin/src/app/feature-modules/events/services/event-pass.service';
+import { IEventPass } from 'projects/shared-models/event-pass.model';
+import { LibToastLogService } from 'projects/shared-services/lib-toastlog.service';
+
+@Component({
+  selector: 'app-entry-pass-scan',
+  templateUrl: './entry-pass-scan.component.html',
+  styleUrls: ['./entry-pass-scan.component.scss'],
+})
+export class EntryPassScanComponent implements OnInit {
+
+  @ViewChild('scanner', { static: false }) scanner: ZXingScannerComponent;
+  @ViewChild('correctSound') correctSound: ElementRef;
+  @ViewChild('incorrectSound') incorrectSound: ElementRef;
+
+  scannerEnabled: boolean = true;
+
+  availableDevices: MediaDeviceInfo[];
+  currentDevice: MediaDeviceInfo = null;
+  hasDevices: boolean;
+  hasPermission: boolean;
+  qrResultString: string;
+  eventPass: IEventPass = null;
+  event_id: string = null;
+
+  formatsEnabled: BarcodeFormat[] = [
+    BarcodeFormat.QR_CODE,
+  ];
+
+  constructor(
+    private eventPassService: EventPassService,
+    private activatedRoute: ActivatedRoute,
+    private toastService: LibToastLogService,
+  ) { }
+
+  ngOnInit(): void {
+    this.activatedRoute.params.subscribe((res) => {
+      this.event_id = res.event_id;
+    })
+  }
+
+  isBoolean(val) {
+    return val === true || val === false;
+  }
+
+  onCodeResult(resultString: string) {
+    this.eventPassService.getEntryPass(this.event_id, resultString).subscribe((data) => {
+      if (!this.isBoolean(data)) {
+        if(!data.attendance){
+          this.correctSound.nativeElement.play();
+          this.eventPass = data;
+        }
+      }
+      else {
+        this.incorrectSound.nativeElement.play();
+        this.toastService.warningDialog('Entry Pass Not found')
+      }
+    })
+  }
+
+  toggleAttendance() {
+    this.eventPassService.toggleAttendance(870).subscribe((res) => {
```

Comment:
What's this 870?

Reply:
this was the event pass id of sample user, forgot to change this to this.eventPass.id